auction-house/program/src/lib.rs

```diff
@@ -966,6 +967,96 @@ pub mod auction_house {
+        receipt_bump: u8,
+        buyer_price: u64,
+        token_size: u64,
+    ) -> ProgramResult {
+        let trade_state = &ctx.accounts.trade_state;
+        let receipt = &mut ctx.accounts.receipt;
+        let auction_house = &ctx.accounts.auction_house;
+        let token_account = &ctx.accounts.token_account;
+        let bookkeeper = &ctx.accounts.bookkeeper;
+        let wallet = &ctx.accounts.wallet;
+
+        let trade_state_bump = trade_state.to_account_info().data.borrow()[0];
+
+        receipt.trade_state = receipt.trade_state;
+        receipt.auction_house = auction_house.key();
+        receipt.token_account = token_account.key();
+        receipt.bookkeeper = bookkeeper.key();
+        receipt.wallet = wallet.key();
+        receipt.buyer_price = buyer_price;
+        receipt.token_size = token_size;
+        receipt.bump = receipt_bump;
+
+        let ts_seeds = [
+            PREFIX.as_bytes(),
+            receipt.wallet.as_ref(),
+            receipt.auction_house.as_ref(),
+            receipt.token_account.as_ref(),
+            auction_house.treasury_mint.as_ref(),
+            token_account.mint.as_ref(),
+            &receipt.buyer_price.to_le_bytes(),
+            &receipt.token_size.to_le_bytes(),
+            &[trade_state_bump],
+        ];
+
+        assert_derivation(
+            &id(),
+            &trade_state.to_account_info(),
+            &ts_seeds,
+        )?;
+
+        Ok(())
+    }
+
+    pub fn burn_receipt<'info>(
+        ctx: Context<'_, '_, '_, 'info, BurnReceipt<'info>>,
+    ) -> ProgramResult {
+        let trade_state = &ctx.accounts.trade_state;
+        let receipt = &mut ctx.accounts.receipt;
+        let auction_house = &ctx.accounts.auction_house;
+        let token_account = &ctx.accounts.token_account;
+        let bookkeeper = &ctx.accounts.bookkeeper;
+
+        let trade_state_bump = trade_state.to_account_info().data.borrow()[0];
+
+        let ts_seeds = [
+            PREFIX.as_bytes(),
+            receipt.wallet.as_ref(),
+            receipt.auction_house.as_ref(),
+            receipt.token_account.as_ref(),
+            auction_house.treasury_mint.as_ref(),
+            token_account.mint.as_ref(),
+            &receipt.buyer_price.to_le_bytes(),
+            &receipt.token_size.to_le_bytes(),
+            &[trade_state_bump],
+        ];
+
+        assert_derivation(
+            &id(),
+            &trade_state.to_account_info(),
+            &ts_seeds,
+        )?;
+
+        if trade_state.data_is_empty() {
+            return Err(ErrorCode::TradeStateDoesntExist.into());
+        }
+
+        let receipt_lamp =  receipt.to_account_info().lamports();
+        **receipt.to_account_info().lamports.borrow_mut() = 0;
+
+        **bookkeeper.to_account_info().lamports.borrow_mut() = receipt
+            .to_account_info()
+            .lamports()
+            .checked_add(receipt_lamp)
+            .ok_or(ErrorCode::NumericalOverflow)?;
```

Comment:
This should be closing out the receipt and send the funds to the wallet that created the receipt. CR here please.

Reply:
need to zero out data.